src/lib/device-online-state.ts

```diff
@@ -0,0 +1,320 @@
+import * as Bluebird from 'bluebird';
+import { DEFAULT_SUPERVISOR_POLL_INTERVAL } from './env-vars';
+import { noop } from 'lodash';
+import { resinApi, root } from '../platform';
+import { captureException } from '../platform/errors';
+import {
+	createPromisifedRedisClient,
+	PromisifedRedisClient,
+} from './redis-promise';
+import * as RedisSMQ from 'rsmq';
+import {
+	REDIS_HOST,
+	REDIS_PORT,
+	API_HEARTBEAT_STATE_TIMEOUT_SECONDS,
+} from './config';
+
+export const getPollIntervalForDevice = (uuid: string): Bluebird<number> => {
+	return resinApi
+		.get({
+			resource: 'device_config_variable',
+			passthrough: {
+				req: root,
+			},
+			options: {
+				$select: 'value',
+				$expand: {
+					device: {
+						$filter: { uuid },
+					},
+				},
+				$filter: {
+					name: {
+						$in: [
+							'BALENA_SUPERVISOR_POLL_INTERVAL',
+							'RESIN_SUPERVISOR_POLL_INTERVAL',
+						],
+					},
+				},
+			},
+		})
+		.then(([pollInterval]: Array<{ value: number }>) => {
```

Comment:
There can be potentially 2 results, which of the two should take priority?

Reply:
Should be resolved now, by ordering the responses and taking the top 1 result 👍